### DISCUSSION TITLE
Feature request: unstyled components should act more like a "headless UI"

### DISCUSSION THREAD

**Discussion by Nickersoft** [saadeghi/daisyui#2099] [Ideas / feature requests]
Hey @saadeghi,

First off - great library! Been using it very extensively in my work. I'm basically using DaisyUI as the building blocks to build my own design system and have overwritten nearly all of the built-in styles that come with it. I'd like to switch over to full "unstyled mode", but I noticed that unstyling some of the components completely removes their usefulness, so-to-speak. A good example of this is the [toggle component](https://daisyui.com/components/toggle/). Removing the styling effectively just turns it into a checkbox, and the styles that make it appear as a toggle are too complex for me to just copy them into my codebase by hand. 

I think it could be superbly helpful to expand the base styles of unstyled components so they act as the most basic form of themselves, similar to a headless UI library, as opposed to just removing _all_ of the CSS. This would make it easier for devs to build design systems on top of Daisy without having to import unnecessary theme CSS that would just end up being overwritten anyway.

**Comment by saadeghi** [comment]
Headless UI only provides the JS functionality (with examples using Tailwind CSS utility classes optionally)  
That way you can use Headless UI with any CSS library or a custom CSS style. Because it's just the JS.

daisyUI's unstyled config on the other hand is trying to provide the "structure" styles without "design decision" styles. So it would be possible to use it as a base for your own custom design system.

In some components, honestly it's hard to draw a line where the structure ends and where the design decision starts but of course I'm willing to fix the issues and improve them as much as possible.

For the toggle, what is your expectations for an unstyled version?  
Do you need a toggle with no colors? What should be the size? (Because I think sizes are design decisions as well)

**Reply by Nickersoft** [reply]
Hey @saadeghi, thanks for the prompt reply! I would expect a toggle with no colors, I suppose (or just B&W or something). The size could be medium by default, with `lg` and `sm` modifiers affecting it as well. Really I think the toggle is the only component that comes to mind that fully breaks when it is unstyled (though I'll keep my eyes open for others). Definitely understand where you're coming from in terms of trying to separate out structure from design decisions.
